scripts/nasa/cmip6_sea_level/process.py

```diff
@@ -0,0 +1,312 @@
+        # are the whole degrees latitude and longitude.  MMM goes from 0-180
+        # while NNN goes from 0-360
+        lstr = str(location)
+        lat = 90 - int(lstr[2:5])
+        lon = int(lstr[6:9])
+        if lon > 180:
+            lon -= 360
+        return prefix + 'grid_1/' + str(lat) + '_' + str(lon)
+
+
+def to_contained_places(location, id2cip):
+    if location == -1:
+        return ''
+    else:
+        dcid = to_place_dcid(location)
+        places = ['dcid:Earth']
+        if dcid in id2cip:
+            places.extend(['dcid:' + x for x in id2cip[dcid]])
+        return ','.join(places)
+
+
+def to_place_type(location):
+    if location == -1:
+        return 'dcs:Place'
+    elif location < 1000000000:
+        return 'dcs:TideGaugeStation'
+    else:
+        return 'dcs:GeoGridPlace_1Deg'
+
+
+def to_sv(quantile, sv_info):
+    if quantile == 0.0:
+        prefix = 'Min'
+    elif quantile == 0.1:
+        prefix = 'Percentile10'
+    elif quantile == 0.5:
+        prefix = 'Median'
+    elif quantile == 0.9:
+        prefix = 'Percentile90'
+    elif quantile == 1.0:
+        prefix = 'Max'
+    else:
+        assert not quantile, 'Unexpected value: ' + str(quantile)
+    return 'dcid:' + prefix + '_' + sv_info[_SVID]
+
+
+def process_statvars(in_file, out_fp, added_svs):
+    sv_info = parse_sv_info(in_file)
+    out_fp.write('# From file ' + _fname(in_file) + '\n\n')
+    for prop, prefix in [('maxValue', 'Max'), ('minValue', 'Min'),
+                         ('medianValue', 'Median'),
+                         ('percentile10', 'Percentile10'),
+                         ('percentile90', 'Percentile90')]:
+        sv_id = prefix + '_' + sv_info[_SVID]
+        prefix_parts = [
+            'Node: dcid:' + sv_id, 'typeOf: dcs:StatisticalVariable',
+            'statType: dcs:' + prop
+        ]
+        if sv_id in added_svs:
+            continue
+        out_fp.write('\n'.join(prefix_parts + sv_info[_PV]))
+        out_fp.write('\n\n')
+        added_svs.add(sv_id)
+
+
+def process_stats(in_file, out_dir):
+    sv_info = parse_sv_info(in_file)
+    ds = xarray.open_dataset(in_file, engine='netcdf4')
+    ncvar = sv_info[_NCVAR]
+    df = ds[ncvar].to_dataframe()
+    df = df.dropna()
+    df = df.reset_index()
+    df = df[(df['quantiles'] == 0.0) | (df['quantiles'] == 0.1) |
+            (df['quantiles'] == 0.5) | (df['quantiles'] == 0.9) |
+            (df['quantiles'] == 1.0)]
+    df['observationAbout'] = df['locations'].apply(
+        lambda x: to_place_dcid(x, 'dcid:'))
+    df['variableMeasured'] = df['quantiles'].apply(lambda x: to_sv(x, sv_info))
+    df['observationDate'] = df['years'].astype(str)
+    df['value'] = df[ncvar]
+    if _UNIT in sv_info:
+        df['unit'] = 'dcs:' + sv_info[_UNIT]
+    if _MMETHOD in sv_info:
+        df['measurementMethod'] = 'dcs:' + sv_info[_MMETHOD]
+    df.drop(['locations', 'years', 'quantiles', ncvar], axis=1, inplace=True)
```

Comment:
alternatively, you can specify which df columns to write to csv

Reply:
I like that better, since its more future proof.